### PR TITLE
disk: define PartitionTableType enum for partition table types

### PR DESCRIPTION
**Pull request by achilleas-k**
This is part of #926 which I'm slowly splitting into smaller, bite-sized PRs.

---

Add a new enum, PartitionTableType, for referring to the two partition table types, "dos" and "gpt".
This is currently not used but will be part of the new custom partition table generator.

### CODE REVIEW

**Review by ondrejbudai — DISMISSED** [review]
Thanks!

**Comment by achilleas-k** [comment]
Needs to be rebased on #1002.

**Review by mvo5 — DISMISSED** [review]
I know you will laugh at me but I would fove a tiny test but otherwise this is fine.

**Comment by achilleas-k** [comment]
> I know you will laugh at me but I would fove a tiny test but otherwise this is fine.

No no, it's my fault.  I should have known and done it already.

**Comment by achilleas-k** [comment]
> I know you will laugh at me but I would fove a tiny test but otherwise this is fine.

Added TWO!

**Review by mvo5 — COMMENTED** [review]
This is very nice, one tiny suggestion inline still (sorry!)

File: pkg/disk/enums_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		expected := disk.PartitionTableType(num)
+
+		assert.NoError(err)
+		assert.Equal(expected, ptt)
+
+		assert.Equal(name, ptt.String())
+	}
+
+	// error test: bad value
+	badPtt := disk.PartitionTableType(3)
```
Comment (mvo5) [inline pkg/disk/enums_test.go:29]: This is very smart! I like this!

**Review by mvo5 — APPROVED** [review]
Thank you!

File: pkg/disk/enums_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	"testing"
+
+	"github.com/osbuild/images/pkg/disk"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnumPartitionTableType(t *testing.T) {
+	enumMap := map[string]uint64{
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply pkg/disk/enums_test.go]: You're absolutely right.  Changed.

**Comment by achilleas-k** [comment]
Merging this because I'm going to need it for #1009.

**Pull request merged**